packages/io/svg-deserializer/index.js

```diff
@@ -131,85 +140,87 @@ const objectify = function (group) {
     i--
   }
 
-  let lnCAG = new CAG()
+  let geometries = []
 
   const params = {
     svgUnitsPmm,
     svgUnitsX,
     svgUnitsY,
     svgUnitsV,
     level,
+    target,
     svgGroups
   }
-  // generate code for all objects
+  // apply base level attributes to all shapes
   for (i = 0; i < group.objects.length; i++) {
     const obj = group.objects[i]
-    let onCAG = shapesMapCsg(obj, objectify, params)
-
-    if ('fill' in obj) {
-    // FIXME when CAG supports color
-    //  code += indent+on+' = '+on+'.setColor(['+obj.fill[0]+','+obj.fill[1]+','+obj.fill[2]+']);\n';
-    }
-    if ('transforms' in obj) {
-      // NOTE: SVG specifications require that transforms are applied in the order given.
-      // But these are applied in the order as required by CSG/CAG
-      let tr = null
-      let ts = null
-      let tt = null
-
-      for (let j = 0; j < obj.transforms.length; j++) {
-        const t = obj.transforms[j]
-        if ('rotate' in t) { tr = t }
-        if ('scale' in t) { ts = t }
-        if ('translate' in t) { tt = t }
-      }
-      if (ts !== null) {
-        const x = ts.scale[0]
-        const y = ts.scale[1]
-        onCAG = onCAG.scale([x, y])
-      }
-      if (tr !== null) {
-        const z = 0 - tr.rotate
-        onCAG = onCAG.rotateZ(z)
-      }
-      if (tt !== null) {
-        const x = cagLengthX(tt.translate[0], svgUnitsPmm, svgUnitsX)
-        const y = (0 - cagLengthY(tt.translate[1], svgUnitsPmm, svgUnitsY))
-        onCAG = onCAG.translate([x, y])
+    let shapes = utils.toArray(shapesMapCsg(obj, objectify, params))
+    shapes = shapes.map((shape) => {
+      if ('transforms' in obj) {
+        // NOTE: SVG specifications require that transforms are applied in the order given.
+        // But these are applied in the order as required by the CSG library
+        let tr
+        let ts
+        let tt
+        for (let j = 0; j < obj.transforms.length; j++) {
+          const t = obj.transforms[j]
+          if ('rotate' in t) { tr = t }
```

Comment:
these need to be changed, we do not have prototype methods anymore in V2
or wait where do the transforms come from in this ?

Reply:
(if they are on the svg objects and not CSG/Jscad, feel free to ignore)